Core/Src/main.c

```diff
@@ -87,14 +89,39 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  __HAL_TIM_SET_COMPARE(&htim1, TIM_CHANNEL_1, 0.05 * __HAL_TIM_GET_AUTORELOAD(&htim1));
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+    /* pot on CH0 of MCP3004 (single input), send bits [1 1 x 0 0] (start, sgl/diff, d2, d1, d0)*/
+    uint8_t tx_data[3] = {0x1, 0x1 << 7, 0x0}; /* pad with zeroes to fit 8 bits */
```

Comment:
You got the correct tx data! Could add `const` to this variable to show this is not supposed to be changed.